### ISSUE TITLE
Relationship between connected members

### ISSUE THREAD

**Issue by ghezibde** [vesta-webtrees-2-custom-modules/vesta_extended_relationships#90]
This is not an issue, rather a suggestion or request.
I work on a webtrees website where several members edit the same and unique tree.
Every week we have several account requests and most requesters don't really understand we're all related and this excellent and useful Vesta Relationship module helps a lot to show that truth.
I'm looking for an option making it straightforward for editors on the tree to show their relationship to each other.
That would give something like this
![image](https://user-images.githubusercontent.com/75539399/212944786-c239feca-9e8c-4860-b6b9-c0e2e73050f3.png)

For such option, it would be  great to be possible to override privacy controls/limitations by showing the only name informations for the steps out of a members privacy perimeters (anyway, this would only concern registered members, thus approved by an admin)

**Comment by ric2016** [comment]
> For such option, it would be great to be possible to override privacy controls/limitations

I don't think this would even be necessary: The list of signed-in users aleready has links to the respective individual pages, even for 'private' users, correct? From the respective urls, you can extract the XRef and enter it into the relationship chart, showing the relationship - if necessary - through private individuals (example: <https://cissee.de/webtrees2/tree/tree/vesta-relationships-1-99/I1825/I400>)

So basically the functiality is already there, you'd just like to have a shortcut for the specific relationship charts - Or am I missing something?

**Comment by ghezibde** [comment]
For sure, we'd need to override the privacy controls for this specific display.
I'd like each connected member to be able to display his own relationship with each of the other connected members from the **_User List_** block.
Most of the listed members are not close enough to him. (most of the members og Ghezibde have a 6, 7, 8 or more relationship while they only have a path length of 4 ou 5 in their profile.
This is an example masquerading a member on ghezibde.
![image](https://user-images.githubusercontent.com/75539399/213009311-070cc927-1b0e-4e2b-a519-47fb7b41d1ed.png)
Difficult to find the right word to explain a possibly mostly different use of webtrees on my side ... since i've been using vesta modules from the beginning and i'm very happy about it (thanks for your work)

**Comment by ric2016** [comment]
I think you have to turn on "Show private relationships" for the respective tree for this chart to work with private individuals.
Control panel > Manage family trees > (the tree) > Privacy

**Comment by ric2016** [comment]
Sure, it makes sense to add an additional button for this. I just wanted to make sure first that nothing else is required.
I'll implement this in Vesta.

**Comment by ghezibde** [comment]
Great !
Thanks

**Comment by ric2016** [comment]
Note that due to limitations of the webtrees module system, the new block will show up under the module name rather than the block name when configuring the blocks on the tree page/my page.

There are no plans to address this in Vesta, it is a general webtrees issue in case a module is used for different parts )tab, chart, block, etc.), and should be improved in webtrees.

The block itself will be displayed on the pages as 'Who is online', just like the original block (plus optional vesta symbol).

<img width="540" alt="mypage" src="https://user-images.githubusercontent.com/18174787/213880391-818a0750-f2bf-409a-87cf-1ecb5aaff2a3.png">

**Comment by ric2016** [comment]
Available in release 2.1.16.1.0

**Comment by ghezibde** [comment]
Option for users relationship chart does not care about options chosen in module setup :

![image](https://user-images.githubusercontent.com/75539399/213912521-97cfed99-e9ee-48f9-9bda-6b4e56109f0d.png)

![image](https://user-images.githubusercontent.com/75539399/213912537-73b53608-260a-4ab8-92d7-0ed90eb92db5.png)

**Comment by ghezibde** [comment]
Block size and looking.
Would it be possible to hide the "Relation with me" sentence and replace it with vesta icon r whatever other one(at list when the option for the icon in the block title is enabled)
The lenght of the block is really longer and because it is often on a second line, it can be really confusing (and a little ugly to me)

![image](https://user-images.githubusercontent.com/75539399/213912694-33bbf129-9eee-4f7e-9313-a3ac32e5aa37.png)

**Comment by ghezibde** [comment]
users update: would you know of a way to "automaticaly" replace the existing block with this new one, even do doing it from the database (sql) ?
if not, i'll have to masquerade each of my 1400 users to make exactly the same change)
Which other way to update users pages would you think of ?

**Comment by ghezibde** [comment]
Some users used their email address as user name.
Connected username appear in the bloc (in the original logged-in users too) and that's not a good thing.
IMO, it would be wiser to cut username after @ (if one) or even better only the real name would be enough in that list ?

**Comment by ric2016** [comment]
> Would it be possible to hide the "Relation with me" sentence

I don't think this would be clearer to users. This is getting too specific - I'll add a css class to the link, then you can style it differently via custom css (see [e.g. here](https://stackoverflow.com/a/47221912)

> Connected username appear in the bloc (in the original logged-in users too) and that's not a good thing.

Same here, probably.

**Comment by ghezibde** [comment]
somes ideas
IMO, users names and users username are two aspects of the same user and we ought to choose which is the more opportune for that list, show both in a limited place is redundant. I can't think about something specific about that ?
An other aspect is about privacy and RGPD : some users use their email address as username and show this information publicly (at least for all connected members) could not be a good thing.

**Comment by ghezibde** [comment]
> > Would it be possible to hide the "Relation with me" sentence
> 

I was meaning : replace the sentence "Relation with me" with an image and Show the message "Relation to  me" as a tooltip, exactly like the envelope and send a message for the the previous link in the same list.

**Comment by ric2016** [comment]
> An other aspect is about privacy and RGPD : some users use their email address as username and show this information publicly (at least for all connected members) could not be a good thing.

In my opinion these kind of issues should be raised in webtrees itself, this is out of scope for this Vesta module. I'm ok with adding the relationship link, but I'm not interested in redesigning the entire block.

**Comment by ric2016** [comment]
> users update: would you know of a way to "automaticaly" replace the existing block with this new one, even do doing it from the database (sql) ?

These settings are stored in 'wt_block'. It seems like replacing the value in column 'module_name' should work. I have not tested this myself - in any case you should backup the table and test this on a separate system first, if possible.

**Comment by ric2016** [comment]
> I was meaning : replace the sentence "Relation with me" with an image and Show the message "Relation to me" as a tooltip, exactly like the envelope and send a message for the the previous link in the same list.

Understood - You will be able to do this via custom css, as described above.

**Comment by ric2016** [comment]
Previous commit has adjustments for:
* Proper chart option selected
* css classes for username and chart link, in order to make these adjustable via custom css

**Comment by ghezibde** [comment]
I'm just having this error (di not applu last commit yet)
![image](https://user-images.githubusercontent.com/75539399/213944398-a486413f-90fa-416d-acae-2cd75b3f7fb9.png)

i think it's due to this case:
a visitor or member can be registered and can log in without having an individual record xref in their profile
after verification in back-office, that's it. I have one or two members registered as visitors and who do not have a individual id in their profile
![image](https://user-images.githubusercontent.com/75539399/213944386-18a6f7d9-9ee0-4834-8489-a1adece191c6.png)

**Comment by ric2016** [comment]
Thanks for reporting, I'll have to fix that.

**Comment by ric2016** [comment]
Previous commit has fix for missinmg individual record xref.

**Comment by ghezibde** [comment]
This option is great, thanks :+1: 
i'm grateful for your attention for my multiple questions.
I was not able to change by a css transformation rule so i just changed $link construction of UserRepositoryExt.php.
I get what i was looking for :
![image](https://user-images.githubusercontent.com/75539399/214244025-d2c891cc-b154-423d-9808-b1e602f85182.png)

but this does totally not satisfy me cause i'll have to change this script for each version update. I'd like to follow your advice and perhaps could your put me on the way (i already searched and looked at many tutos on the net).
Do i have to know which css container is concerned ?
If possible for you, would you have an example of what you suggest so i can write what i'm looking for from that example.

**Comment by ric2016** [comment]
> Do i have to know which css container is concerned ?

Of course, how else would you be able to apply the rule? In general, use the browser developer tools to investigate the respective element. An example that replaces the text:

```
a.vesta-relationship-chart-link {
  font-size: 0;
}

a.vesta-relationship-chart-link:after {
  content: 'new text';
  font-size: initial;
}
```

Adding an image may be a bit trickier, but should be possible as well.
